### PR TITLE
Upgrade Helm and other exekube bits

### PR DESCRIPTION
**Pull request by stepanstipl**
This PR:
- Updates Helm to 2.12 (this is a requirement for Istio work)
- Updates Alpine to 2.8 and Go to 1.11
  - includes fixes to build and run `google-cloud-monitoring` and `-logging` gems, which depend on `grpc` and `protobuf` and their native extensions (see  https://github.com/protocolbuffers/protobuf/issues/4460, https://github.com/protocolbuffers/protobuf/issues/2335, https://github.com/gliderlabs/docker-alpine/issues/424, https://github.com/grpc/grpc/issues/6525 for details)
- Update Terraform to 0.11.10 and Terragrunt to 0.17.3
- Moves Terragrunt cache to /tmp - this prevents lot of `.terragrung-cache` dirs in local environments and some issues after upgrading Terraform provider versions
- Updates kubectl to 1.11
- Updates GCP cloud SDK to 228
- Updates all Terrafrom providers to their latest versions (as of today)

Depends on https://github.com/gpii-ops/gpii-infra/pull/242

### CODE REVIEW

File: dockerfiles/alicloud/Dockerfile [hunk 1]
```diff
@@ -16,15 +16,17 @@ ADD ./cli ./src/github.com/exekube/exekube/cli
...
-ENV KUBECTL_VERSION 1.9.3
-ENV TERRAFORM_VERSION 0.11.5
-ENV TERRAGRUNT_VERSION 0.14.6
+ENV HELM_VERSION 2.12.0
+ENV KUBECTL_VERSION 1.11.5
+ENV TERRAFORM_VERSION 0.11.10
+ENV TERRAGRUNT_VERSION 0.17.3
 ENV TERRAFORM_PROVIDER_HELM_VERSION 0.6.2
 ENV TERRAFORM_PROVIDER_ALICLOUD_VERSION 1.9.1
+# Terragrun tcache directory
```
Comment (mrtyler) [inline dockerfiles/alicloud/Dockerfile]: Misplaced space

File: dockerfiles/aws/Dockerfile [hunk 2]
```diff
@@ -20,16 +20,18 @@ RUN go build -o ./bin/xk github.com/exekube/exekube/cli
...
-ENV KUBECTL_VERSION 1.10.5
-ENV HELM_VERSION 2.9.1
-ENV TERRAFORM_VERSION 0.11.7
-ENV TERRAGRUNT_VERSION 0.14.10
+ENV KUBECTL_VERSION 1.11.5
+ENV HELM_VERSION 2.12.0
+ENV TERRAFORM_VERSION 0.11.10
+ENV TERRAGRUNT_VERSION 0.17.3
 ENV TERRAFORM_PROVIDER_HELM_VERSION 0.6.2
+# Terragrun tcache directory
```
Comment (mrtyler) [inline dockerfiles/aws/Dockerfile]: Misplaced space

File: dockerfiles/google/Dockerfile [hunk 3]
```diff
@@ -20,16 +20,18 @@ RUN go build -o ./bin/xk github.com/exekube/exekube/cli
...
-ENV TERRAGRUNT_VERSION 0.14.10
+ENV CLOUD_SDK_VERSION 228.0.0
+ENV KUBECTL_VERSION 1.11.5
+ENV HELM_VERSION 2.12.0
+ENV TERRAFORM_VERSION 0.11.10
+ENV TERRAGRUNT_VERSION 0.17.3
 ENV TERRAFORM_PROVIDER_HELM_VERSION 0.6.2
 ENV GOOGLE_CLOUD_MONITORING_VERSION 0.29.2
 ENV GOOGLE_CLOUD_LOGGING_VERSION 1.5.7
+# Terragrun tcache directory
```
Comment (mrtyler) [inline dockerfiles/google/Dockerfile]: Misplaced space

File: dockerfiles/google/Dockerfile [hunk 4]
```diff
@@ -52,6 +54,9 @@ RUN apk --no-cache add \
         g++ \
         make \
         ruby-dev && \
+    # Following is needed for various native exptension issues on Alpine with protobuf and grpc plugins
```
Comment (mrtyler) [inline dockerfiles/google/Dockerfile]: extension

Comment (mrtyler) [inline dockerfiles/google/Dockerfile]: This is a little vague, so I might link to the most relevant issue(s) here, or back to the description of this PR.

**Comment by natarajaya** [comment]
Please fix typos and then LGTM.

File: dockerfiles/google/Dockerfile (hunk 4 above)
Comment (stepanstipl) [reply dockerfiles/google/Dockerfile]: Thanks. So there are like 4 different issues I've hit and each of them requires some part of the fix and if not present will lead to different issue(s):
- doing the `gem update`
- explicitly installing `google-protobuf` with `ruby` platfotm
- same for `grpc` and it's dependency `etc`
- pinning down the protobuf version at `3.6.1`

I think thiat is a bit too long to explain in the code. All of the issue(s) are referenced in the commit message (as well as in the PR description), so I'll remove the comment here.

**Comment by stepanstipl** [comment]
Thanks for reviews @mrtyler and @natarajaya - I've fixed the typos and removed the comment from the code (detailed commit message contains references to the relevant issues).

**Pull request merged**